### DISCUSSION TITLE
Confusion regarding upper right ice-pico pmod

### DISCUSSION THREAD

**Discussion by MrJake222** [tinyvision-ai-inc/pico-ice#53] [General]
According to [this](https://github.com/tinyvision-ai-inc/pico-ice/blob/main/Docs/pinout.md#spi-pinouts) the "Upper right PMOD" pins are connected to the SPI bus. However the schematic shows that the PMOD and SPI bus get their own pins:
![image](https://github.com/user-attachments/assets/c3eb74d2-e4a6-4dba-bbcd-c0331f59858d)
![image](https://github.com/user-attachments/assets/4e97fef3-2e1f-4fdb-a910-ca26434aad37)

According to SPI Pinouts docs here, for example pin 20 would be shorted with pin 12?
![conf](https://github.com/user-attachments/assets/4feae9f2-fab3-491f-90c7-5b718e749198)

Also this would break second UART forwarding and be really limiting in ice-pico communications when the flash is in use after the bitstream is read?

**Comment by josuah** [comment]
I notice that I never came back to answer this. Sorry about that.

Yes you are right, there are some inconsistencies. For instance, `ICE_FLASH_IO2` that you are pointing at is not brought to a Pmod:

![scrot_20250215_155145_1326x536](https://github.com/user-attachments/assets/c1200cff-4eee-4072-8943-dbad0efe495a)

These must have been debug signals added by the author, exporting them to FPGA signals through the RTL design rather than representing the interconnections at the PCB level.

This will need some verification.

Thanks for pointing it!
